src/main/java/edu/wisc/my/messages/service/ExpiredMessagePredicate.java

```diff
@@ -27,7 +28,8 @@ public boolean test(Message message) {
       new IsoDateTimeStringBeforePredicate(when);
 
     try {
-      return beforeWhen.test(message.getExpireDate());
+      MessageFilter filter = message.getFilter();
+      return beforeWhen.test(filter.getExpireDate());
```

Comment:
Problem: doesn't handle the case where the `message` doesn't have a `filter`.

I think this is why I get

```
{
  "messages": []
}
```

when requesting `/messages` on this branch - the message with no `filter` seems to be expired rather than never-expiring.